sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -471,17 +479,30 @@ abstract class SparkPlanMeta[INPUT <: SparkPlan](plan: INPUT,
       }
     } else if (exchanges.isEmpty) {
       // we only have query stages, and we cannot do anything to modify them at this point
-      // since they already started to execute
+      // since they already started to execute, but we verify that they are both on CPU or
+      // both on GPU
+      if (queryStages.map(isGpuQueryStage).distinct.size == 2) {
```

Comment:
so for my edification, we can't get in this situation unless there is a bug, or perhaps a change in behavior from AQE rules proper?